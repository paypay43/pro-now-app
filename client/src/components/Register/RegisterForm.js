import React, { useState } from 'react';
import TokenService from '../../services/token-service';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import AuthApiService from '../../services/auth-api-service';
import './register.css';

export default function RegisterForm(props) {
  const [error, setError] = useState(null);

  /*const handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    ;
  };*/

  const submitForm = e => {
    e.preventDefault();
    setError(null);
    const { username, password, confirm_password, email } = e.target;

    if (password.value !== confirm_password.value) {
      setError({ ...error, error: 'Passwords do not match' });
      return;
    }

    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      email: email.value
    })
      .then(user => {
        username.value = '';
        password.value = '';
        confirm_password.value = '';
        email.value = '';
        props.onRegistrationSuccess();
      })
      .catch(res => {
        setError({ error: res.error });
      });
  };

  const displayError = () => {
    return Object.keys(error).map(e => {
      return (
        <span key={e}>
          {e}: {error[e]}
        </span>
      );
    });
  };

  return (
    <form onSubmit={submitForm} className="register-form app-form">
      <h1>Register</h1>
      <div role="alert">{error && <p className="red">{displayError()}</p>}</div>
      <label className="label-control" htmlFor="username">
        Username
      </label>
      <input
        name="username"
        type="username"
        id="inputUsername"
        className="form-control"
        required
        autoFocus=""
      />
      <label className="label-control" htmlFor="password">
        Password
      </label>
      <input
        name="password"
        type="password"
        id="inputPassword"
        className="form-control"
        required
      />
      <label className="label-control" htmlFor="confirm_password">
        Confirm password
      </label>
      <input
        name="confirm_password"
        type="password"
        id="inputPasswordConfirm"
        className="form-control"
        required
      />
      <label className="label-control" htmlFor="email">
        Email
      </label>
      <input
        name="email"
        type="email"
        id="inputEmail"
        className="form-control"
        required
      />
      <button className="btn" type="submit">
        Register
      </button>
    </form>
  );
}
