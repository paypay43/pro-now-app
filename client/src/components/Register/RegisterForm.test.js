import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegisterForm from './RegisterForm';

describe(`RegisterForm component`, () => {
  it('renders register form', () => {
    const wrapper = shallow(<RegisterForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegisterForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
