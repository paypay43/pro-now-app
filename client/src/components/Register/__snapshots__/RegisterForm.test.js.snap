// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders register form 1`] = `
<form
  className="register-form app-form"
  onSubmit={[Function]}
>
  <h1>
    Register
  </h1>
  <div
    role="alert"
  />
  <label
    className="label-control"
    htmlFor="username"
  >
    Username
  </label>
  <input
    autoFocus=""
    className="form-control"
    id="inputUsername"
    name="username"
    required={true}
    type="username"
  />
  <label
    className="label-control"
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="form-control"
    id="inputPassword"
    name="password"
    required={true}
    type="password"
  />
  <label
    className="label-control"
    htmlFor="confirm_password"
  >
    Confirm password
  </label>
  <input
    className="form-control"
    id="inputPasswordConfirm"
    name="confirm_password"
    required={true}
    type="password"
  />
  <label
    className="label-control"
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="form-control"
    id="inputEmail"
    name="email"
    required={true}
    type="email"
  />
  <button
    className="btn"
    type="submit"
  >
    Register
  </button>
</form>
`;
